src/app/components/LayerPicker/LayerMenu.tsx

```diff
@@ -51,78 +52,81 @@ export const DisabledLayerMenuItem: FC<BaseLayerMenuItemProps> = ({ divider, lay
 
 type LayerMenuItemProps = LayerMenuProps &
   BaseLayerMenuItemProps & {
-    hoveredLayer?: Layer
-    setHoveredLayer: (layer?: Layer) => void
+    hoveredScope?: SearchScope
+    setHoveredScope: (scope?: SearchScope) => void
   }
 
 export const LayerMenuItem: FC<LayerMenuItemProps> = ({
-  activeLayer,
   divider,
-  layer,
-  network,
-  selectedLayer,
-  selectedNetwork,
-  setHoveredLayer,
-  setSelectedLayer,
+  targetScope,
+  selectedScope,
+  setHoveredScope,
+  setSelectedScope,
 }) => {
   const { t } = useTranslation()
   const labels = getLayerLabels(t)
-  const activeLayerSelection = layer === selectedLayer
+  const activeScope = useScopeParam()!
+  const isSelected =
+    targetScope.network === selectedScope?.network && targetScope.layer === selectedScope?.layer
+  const isActive = targetScope.network === activeScope.network && targetScope.layer === activeScope.layer
 
   return (
     <MenuItem
       divider={divider}
       onMouseEnter={() => {
-        setHoveredLayer(layer)
+        setHoveredScope(targetScope)
       }}
       onMouseLeave={() => {
-        setHoveredLayer()
+        setHoveredScope()
       }}
       onClick={() => {
-        setSelectedLayer(layer)
+        setSelectedScope(targetScope)
       }}
-      selected={activeLayerSelection}
-      tabIndex={activeLayerSelection ? 0 : -1}
+      selected={isSelected}
+      tabIndex={isSelected ? 0 : -1}
     >
       <ListItemText>
-        {labels[layer]}
-        {selectedNetwork === network && activeLayer === layer && (
-          <LayerMenuItemCaption>{t('layerPicker.selected')}</LayerMenuItemCaption>
-        )}
+        {labels[targetScope.layer]}
+        {isActive && <LayerMenuItemCaption>{t('layerPicker.active')}</LayerMenuItemCaption>}
       </ListItemText>
-      {layer === selectedLayer && <KeyboardArrowRightIcon />}
+      {isSelected && <KeyboardArrowRightIcon />}
     </MenuItem>
   )
 }
 type LayerMenuProps = {
-  activeLayer: Layer
-  network: Network
-  selectedLayer?: Layer
-  selectedNetwork: Network
-  setSelectedLayer: (layer: Layer) => void
+  selectedNetwork?: Network
+  selectedScope?: SearchScope
+  setSelectedScope: (scope: SearchScope) => void
 }
 
-export const LayerMenu: FC<LayerMenuProps> = ({
-  activeLayer,
-  network,
-  selectedLayer,
-  selectedNetwork,
-  setSelectedLayer,
-}) => {
-  const currentScope = useScopeParam()
-  const [hoveredLayer, setHoveredLayer] = useState<undefined | Layer>()
-  const options = Object.values(Layer)
+type LayerMenuOption = {
+  scope: SearchScope
+  enabled: boolean
+}
+
+const getOptionsForNetwork = (network: Network, activeScope?: SearchScope | undefined): LayerMenuOption[] =>
```

Comment:
`activeScope?: SearchScope | undefined`
this parameter is never provided?

Reply:
Oops, that was accidentally dropped. But we need is so I added it back now.. Thanks, good catch.